System/Random/MWC/Distributions.hs

```diff
@@ -355,3 +357,133 @@ pkgError func msg = error $ "System.Random.MWC.Distributions." ++ func ++
 --   (2007). Gaussian random number generators.
 --   /ACM Computing Surveys/ 39(4).
 --   <http://www.cse.cuhk.edu.hk/~phwl/mt/public/archives/papers/grng_acmcs07.pdf>
+--
+-- * Kachitvichyanukul, V. and Schmeiser, B. W.  Binomial Random
+--   Variate Generation.  Communications of the ACM, 31, 2 (February,
+--   1988) 216. <https://dl.acm.org/doi/pdf/10.1145/42372.42381>
+
+-- | Random variate generator for Binomial distribution
+--
+-- The probability of getting exactly k successes in n trials is
+-- given by the probability mass function:
+--
+-- \[
+-- f(k;n,p) = \Pr(X = k) = \binom n k  p^k(1-p)^{n-k}
+-- \]
+binomial :: forall g m . StatefulGen g m
+         => Int               -- ^ Number of trials
+         -> Double            -- ^ Probability of success (returning True)
+         -> g                 -- ^ Generator
+         -> m Int
+{-# INLINE binomial #-}
+binomial nTrials prob gen =
+  if prob == 0.0
```

Comment:
Purely stylistically. Could you turn nested ifs into guards?

Reply:
Yes done :-)